### ISSUE TITLE
defineExpose not working with <StoryblokComponent>

### ISSUE THREAD

**Issue by Dollique** [storyblok/storyblok-vue#403]
### TL;DR
Using defineExpose() on a child component does not work if the parent uses <StoryblokComponent>.

### Description
If I want my parent component to access a method from my child component I can use defineExpose() to expose the method to the parent.

**Child Component**

```
defineExpose({
  myMethod,
})
```

**Parent Component**
use `ref` like this:

```
<StoryblokComponent
   ref="childComponent"
  :blok="blok"
/>
```

And the method can be called like this from the parent:

```
const childComponent = ref([]);
childComponent.value.myMethod()
```

### Workaround
It works using the vue `<component>` wrapper instead of `<StoryblokComponent>`.

### Reproduction

https://stackblitz.com/edit/sb-nuxt-3-define-expose?file=storyblok%2FGrid.vue

### Steps to reproduce

In my stackblitz the Grid is the parent and the Feature component the child.

Open the StackBlitz result in a new Tab and open console to see the outputs.
You will see myMethod() is undefined and not working.

If you use <component> instead (already prepared in Grid.vue) it works.

Note: If you use both `<component>` and `<StoryblokComponent>` strangely both work...?

### System Info

```
shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    Memory: 3.07 GB / 15.92 GB
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.21.1 - ~\node_modules\.bin\yarn.CMD
    npm: 8.3.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.2215.0), Chromium (116.0.1938.54)
    Internet Explorer: 11.0.22621.1
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://www.storyblok.com/trust-center#code-of-conduct?utm_source=github.com&utm_medium=readme&utm_campaign=storyblok-js-client)

**Comment by alvarosabu** [comment]
Hi, @Dollique Thanks for taking the time to open the issue. Effectively `StoryblokComponent` does not expose its context by design https://github.com/storyblok/storyblok-vue/blob/main/lib/StoryblokComponent.vue so this is likely to be a feature request.

To prioritize it accordingly, I would need to understand the practical use case of exposing the context via `templateRefs` vs, for example, sharing a state or methods via composables or by using `<component :is />` directly ? The reason I am asking this is because using this approach will tightly coupled both parent and child component.

A workaround would be to use `component :is="blok.component" />` directly which is basically what is under the hood of `StoryblokComponent`. That way you would not need to pass up several `defineExpose`

**Comment by Dollique** [comment]
My current use case is an Accordion component with a AccordionItem child-component. The AccordionItem component has a method `closeItem()` to close itself. Now in my Accordion there is another handler called `onCloseAll()`, where I loop through all the AccordionItem childs and call `closeItem()` on each.

Right now my code looks like this.

**Accordion.vue (parent)**

```
<template>
  <section v-editable="blok" class="accordion">
    <!-- TODO: change to <StoryblokComponent when issue is fixed: https://github.com/storyblok/storyblok-nuxt/issues/537 -->
    <component
      :is="childblok.component"
      v-for="childblok in blok.item"
      :key="childblok._uid"
      ref="childAccordion"
      :blok="childblok"
      @closeAll="onCloseAll"
    />
  </section>
</template>

<script setup lang="ts">
defineProps<{
  blok: {
    type: Object
    required: true
  }
}>()

const childAccordion = ref([])

const onCloseAll = function () {
  for (let [index, child] of childAccordion.value.entries()) {
    // check if closeItem function exists
    if (typeof childAccordion.value[index].closeItem === 'function') {
      childAccordion.value[index].closeItem()
    }
  }
}
</script>
```

**AccordionItem (child)**

```
<template>
  <section v-editable="blok" class="accordion-item" ref="accordionItem">
    <h2 @click="slidetoggle">{{ blok.title }}</h2>
    <div class="accordion-item__content">
      <div v-html="renderedRichText"></div>
    </div>
  </section>
</template>

<script setup lang="ts">
const props = defineProps<{
  blok: {
    type: Object
    required: true
  }
}>()

const emit = defineEmits(['closeAll'])

const renderedRichText = computed(() => renderRichText(props.blok.content))
const accordionItem = ref(null)

const slidetoggle = function () {
  const el = accordionItem.value.querySelector('.accordion-item__content')
  const ch = el.clientHeight
  const sh = el.scrollHeight
  const isCollapsed = !ch
  const noHeightSet = !el.style.height

  if (isCollapsed || noHeightSet) {
    emit('closeAll') // close all open items

    el.style.height = sh + 'px'
    accordionItem.value.classList.add('active')
  } else {
    el.style.height = 0 + 'px'
    accordionItem.value.classList.remove('active')
  }

  if (noHeightSet) return slidetoggle.call(accordionItem)
}

const closeItem = function () {
  const el = accordionItem.value.querySelector('.accordion-item__content')

  el.style.height = 0 + 'px'
  accordionItem.value.classList.remove('active')
}

defineExpose({
  accordionItem,
  closeItem,
})
</script>
```

As you can see currently when an AccordionItem is closed, `slidetoggle()` is called which then emits `closeAll()` to the parent Accordion.

I am aware this couples the components, but I think in this case I'd argue that's okay.
The alternative approach here could be using state for each Accordion (probably save each item in an array?). I wanted to keep my code simple, so `defineExpose()` seemed like the best option.

I can keep the code like this using `<component>`. The reason I wanted to use `<StoryblokComponent>` is to have similar code in all my Storyblok components.

**Comment by alvarosabu** [comment]
It absolutely makes sense, I will work on it :D
